Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/Receivers/InteractionReceiver.cs

```diff
@@ -0,0 +1,415 @@
+        private List<GameObject> interactables = new List<GameObject>();
+
+        public List<GameObject> Interactables
+        {
+            get { return interactables; }
+            set { value = interactables; }
+        }
+
+        /// <summary>
+        /// List of linked targets that the receiver affects
+        /// </summary>
+        [Tooltip("Targets for the receiver to ")]
+        private List<GameObject> targets = new List<GameObject>();
+
+        public List<GameObject> Targets
+        {
+            get { return targets; }
+            set { value = targets; }
+        }
+
+        [Tooltip( "If true, this object will remain the prime focus while select is held" )]
+        [SerializeField]
+        private bool lockFocus = false;
+
+        /// <summary>
+        /// Flag for locking focus while selected
+        /// </summary>
+        public bool LockFocus
+        {
+            get
+            {
+                return lockFocus;
+            }
+            set
+            {
+                lockFocus = value;
+                CheckLockFocus(_selectingFocuser);
+            }
+        }
+        
+        #endregion
+
+        #region Private and Protected Members
+
+        /// <summary>
+        /// Protected focuser for the current selecting focuser
+        /// </summary>
+        protected IMixedRealityPointer _selectingFocuser;
+        
+        #endregion
+
+        /// <summary>
+        /// On start subscribe to all interaction events on elements in the interactables list.
+        /// </summary>
+        protected override void OnEnable()
+        {
+            base.OnEnable();
+            //Still need to register since focus isn't required
+            if(InputSystem != null && InputSystem.IsInputEnabled)
+            {
+                InputSystem.Register( gameObject );
+            }
+        }
+
+        /// <summary>
+        /// On disable remove all linked interactables from the delegate functions
+        /// </summary>
+        protected override void OnDisable()
+        {
+            //Clean up our event subscription
+            if ( InputSystem != null && InputSystem.IsInputEnabled )
+            {
+                InputSystem.Unregister( gameObject );
+            }
+            base.OnDisable();
+        }
+
+        /// <summary>
+        /// Register an interactable with this receiver.
+        /// </summary>
+        /// <param name="interactable">takes a GameObject as the interactable to register.</param>
+        public virtual void Registerinteractable(GameObject interactable)
+        {
+            if (interactable == null || interactables.Contains(interactable))
+            {
+                return;
+            }
+            interactables.Add(interactable);
+        }
+
+#if UNITY_EDITOR
+        /// <summary>
+        /// When selected draw lines to all linked interactables
+        /// </summary>
+        protected virtual void OnDrawGizmosSelected()
```

Comment:
`OnDrawGizmosSelected` only gets called in the editor anyway. Do we have editor specific code in this ifdef?